benchmarks/segmentation_layers.py

```diff
@@ -1,175 +1,329 @@
-    print(utils.benchmark_fn(batch_norm_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(max_pool2d_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(max_pool2d_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(cross_entropy_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(cross_entropy_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(dropout_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(dropout_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(interpolate_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(interpolate_nt_tensors_same_size(), warmup=2.0))
+        self.H = int(args[2])
+        print("H = ", self.H)
+
+        self.W = int(args[3])
+        print("W = ", self.W)
+        
+        self.delta = float(args[4])
+        print("delta = ", self.W)
+
+        self.warmup = float(args[5])
+        print("warmup = ", self.W)
+        
+        self.conv2d = torch.nn.Conv2d(self.C, 3, kernel_size= (1, 1), bias=False)
+        self.batch_norm = torch.nn.BatchNorm2d(self.C, 1e-05, 0.1)
+        self.batch_norm.eval()
+        self.max_pool2d = torch.nn.MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=(0, 0), dilation=(1, 1))
+
+    def get_max_size(self, obj, res=None):
+        if res is None:
+            res = [1]
+
+        if isinstance(obj, list) or isinstance(obj, tuple):
+            for o in obj:
+                res = self.get_max_size(o, res)
+
+        if isinstance(obj, nestedtensor.nested.nested.NestedTensor):
+            tres = self.get_max_size(obj.unbind())
+            while len(tres) > len(res):
+                    res.append(0)
+
+            res = [max(i, j) for (i, j) in zip(res, tres)]
+
+        if isinstance(obj, torch.Tensor):
+            # scalar
+            if obj.dim() == 0 and obj.numel() == 1:
+                res = [1]
+            else:
+                while len(obj.size()) > len(res):
+                    res.append(0)
+
+                res = [max(i, j) for (i, j) in zip(res, obj.size())]
+
+        return res
+
+    def pad_tensor_to_shape(self, t, goal_shape):
+        padd = ()
+        tup = tuple(t.size())
+        assert(t.dim() == len(goal_shape))
+        for i in range(len(tup)):
+            padd = (0, goal_shape[i] - tup[i]) + padd
+        new_tensor = F.pad(t, padd)
+        new_tensor = new_tensor.reshape(goal_shape)
+        return new_tensor
+
+    def pad_tensors_to_max_shape(self, inputs):
+        max_shape = self.get_max_size(inputs)
+        padded_inputs = [self.pad_tensor_to_shape(t, max_shape) for t in inputs]
+        return padded_inputs
+
+    def get_input(self, return_targets=False, pad=False):
+        inputs = []
+        targets = []
+        for i in range(self.N):
+            h_change = random.randint(self.delta * 10 * (-self.H), self.delta * 10 * self.H) / 10
```

Comment:
This is closer to what I was suggesting. The only issue with randint is that it's not repeatable. Of course in order to track improvements we need to be able to run the exact same benchmark multiple times.